compiler+runtime/src/cpp/jank/runtime/module/loader.cpp

```diff
@@ -338,47 +339,101 @@ namespace jank::runtime::module
       return err(fmt::format("unable to find module: {}", module));
     }
 
-    string_result<void> res{ err(fmt::format("no sources for registered module: {}", module)) };
+    auto const with_module_type{ [&](module_type const type) {
+      loader::find_result res{ entry->second, type };
+      return res;
+    } };
 
     //fmt::println("loading nested module {}", module);
 
-    /* When we're compiling, we always load from source. Otherwise, we
-     * load from source if we specifically chose to do so. */
-    bool const compiling{ truthy(rt_ctx.compile_files_var->deref()) };
-    if(compiling || ori == origin::source)
+    if(ori == origin::source)
     {
       if(entry->second.jank.is_some())
       {
-        res = load_jank(entry->second.jank.unwrap());
+        return with_module_type(module_type::jank);
       }
       else if(entry->second.cljc.is_some())
       {
-        res = load_cljc(entry->second.cljc.unwrap());
+        return with_module_type(module_type::cljc);
       }
     }
     else
     {
-      /* TODO: origin must be latest, so we need to find that. Even if there
-       * is a binary, we need to only choose it if it's as new, or newer, than
-       * the source. */
-      if(entry->second.o.is_some())
+      if(entry->second.o.is_some()
+         && (entry->second.jank.is_some() || entry->second.cljc.is_some()))
       {
-        res = load_o(module, entry->second.o.unwrap());
+        auto const o_file_path{ native_transient_string{ entry->second.o.unwrap().path } };
+
+        native_transient_string source_path{};
+        module_type module_type{};
+
+        if(entry->second.jank.is_some())
+        {
+          source_path = entry->second.jank.unwrap().path;
+          module_type = module_type::jank;
+        }
+        else
+        {
+          source_path = entry->second.cljc.unwrap().path;
+          module_type = module_type::cljc;
+        }
```

Comment:
You're not taking into account that JAR usecase here. Each file entry has two fields (`archive_path` and `path`). Only one of them will be set, but it won't always be `path`.

Reply:
If the jank/cljc souces are in archive, how do we determine the modified time of these files?

I think we can just consider jar's modified time as the source's start time because they will only be used as a dep?

Is there any other case, I am missing?

Reply:
> I think we can just consider jar's modified time as the source's start time because they will only be used as a dep?

Exactly, just the JAR's time.